### PR TITLE
EVG-16763 include task timed out in trigger failures

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-16763](https://jira.mongodb.org/browse/EVG-16763)

### Description 
Need to include task timed out for triggers.

### CODE REVIEW

File: trigger/task.go [hunk 1]
```diff
@@ -703,8 +703,11 @@ func (t *taskTriggers) taskRuntimeChange(sub *event.Subscription) (*notification
 	return t.generate(sub, fmt.Sprintf("changed in runtime by %.1f%% (over threshold of %s%%)", percentChange, percentString), "")
 }
 
-func isFailedTaskStatus(status string) bool {
-	return status == evergreen.TaskFailed || status == evergreen.TaskSystemFailed || status == evergreen.TaskTestTimedOut
+// isValidFailedTaskStatus only matches task statuses that should be triggered for failure.
+// For example, it excludes  setup failures.
+func isValidFailedTaskStatus(status string) bool {
```
Comment (malikchaya2) [inline trigger/task.go:708]: I feel like the addition of the word `valid` here doesn't really help me understand better what the function does. I personally feel like isFailedTaskStatus with the comment you added feels sufficient to me. 
But I don't feel strongly either way so you decide :)

**Review by malikchaya2 — APPROVED** [review]

File: trigger/task.go [hunk 2]
```diff
@@ -703,8 +703,11 @@ func (t *taskTriggers) taskRuntimeChange(sub *event.Subscription) (*notification
...
+	return status == evergreen.TaskFailed || status == evergreen.TaskSystemFailed ||
```
Comment (bynn) [inline trigger/task.go:709]: would it make sense to use [TaskNonGenericFailureStatuses](https://github.com/evergreen-ci/evergreen/blob/7fa53940bcca04122f7c23e7fffcdfa37a1139df/globals.go#L314)?

**Review by bynn — APPROVED** [review]
if you see a notification from me it's because i asked a question and deleted it!

File: trigger/task.go (hunk 1 above)
Comment (ablack12) [reply trigger/task.go:708]: I kind of just want to distinguish it from the non-private-trigger function "IsFailedTaskStatus" but yeah i recognize it doesn't add much

Comment (malikchaya2) [reply trigger/task.go:708]: Oh, that makes sense

**Pull request merged**